mirall-1.2.1: CMake Error at doc/cmake_install.cmake:36 (FILE): file INSTALL cannot find "/root/mirall-build/doc/html".

Hello,

running make install after a successfull build, i get the following error:

loki mirall-build # make install
[ 17%] Built target owncloudsync
[ 58%] Built target mirall
[100%] Built target owncloud
Install the project...
-- Install configuration: "Debug"
-- Installing: /usr/local/etc/sync-exclude.lst
-- Installing: /usr/local/lib64/libowncloudsync.so.1.2.1
-- Installing: /usr/local/lib64/libowncloudsync.so.0
-- Installing: /usr/local/lib64/libowncloudsync.so
-- Removed runtime path from "/usr/local/lib64/libowncloudsync.so.1.2.1"
-- Installing: /usr/local/share/applications/owncloud.desktop
-- Installing: /usr/local/share/icons/hicolor/48x48/apps/mirall.png
-- Installing: /usr/local/share/icons/hicolor/48x48/apps/owncloud.png
-- Installing: /usr/local/share/mirall/i18n/mirall_lb.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_ko.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_lv.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_he.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_en.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_da.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_sl.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_et_EE.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_pt_PT.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_fi_FI.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_th_TH.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_eo.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_fa.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_gl.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_ru_RU.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_it.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_ms_MY.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_cs_CZ.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_sr@latin.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_vi.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_tr.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_mk.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_zh_CN.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_sk_SK.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_el.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_zh_TW.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_sv.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_nb_NO.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_hr.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_eu.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_uk.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_si_LK.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_nl.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_pl.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_ca.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_fr.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_lt_LT.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_es_AR.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_hu_HU.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_es.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_oc.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_ta_LK.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_ja_JP.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_pt_BR.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_ru.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_de.qm
-- Installing: /usr/local/share/mirall/i18n/mirall_ro.qm
-- Installing: /usr/local/bin/mirall
-- Removed runtime path from "/usr/local/bin/mirall"
-- Installing: /usr/local/bin/owncloud
-- Removed runtime path from "/usr/local/bin/owncloud"
CMake Error at doc/cmake_install.cmake:36 (FILE):
file INSTALL cannot find "/root/mirall-build/doc/html".
Call Stack (most recent call first):
cmake_install.cmake:42 (INCLUDE)

make: **\* [install] Fehler 1

Regards,
Norman

--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -14,11 +14,13 @@ if(SPHINX_FOUND)
 
     configure_file("${CMAKE_CURRENT_SOURCE_DIR}/conf.py.in" conf.py @ONLY)
 
-    add_custom_target(doc DEPENDS doc-html doc-man COMMENT "Building documentation...")
-    if (UNIX AND (NOT APPLE))
-        install(DIRECTORY ${SPHINX_HTML_DIR} DESTINATION ${CMAKE_INSTALL_DOCDIR})
-        install(DIRECTORY ${SPHINX_MAN_DIR} DESTINATION ${CMAKE_INSTALL_MANDIR})
-    endif (UNIX AND (NOT APPLE))
+    if(WITH_DOC)
+       add_custom_target(doc ALL DEPENDS doc-html doc-man COMMENT "Building documentation...")
+       install(DIRECTORY ${SPHINX_HTML_DIR} DESTINATION ${CMAKE_INSTALL_DOCDIR})
+       install(DIRECTORY ${SPHINX_MAN_DIR} DESTINATION ${CMAKE_INSTALL_MANDIR})
+    else(WITH_DOC)
+       add_custom_target(doc DEPENDS doc-html doc-man COMMENT "Building documentation...")
+    endif(WITH_DOC)
 
     if(PDFLATEX_FOUND)
         # if this still fails on Debian/Ubuntu, run
@@ -32,9 +34,9 @@ if(SPHINX_FOUND)
         add_custom_target(doc-pdf make -C ${SPHINX_PDF_DIR} all-pdf
 		                  DEPENDS doc-latex )
         add_dependencies(doc doc-pdf)
-        if (UNIX AND (NOT APPLE))
-	    install(DIRECTORY ${SPHINX_PDF_DIR} DESTINATION ${CMAKE_INSTALL_DOCDIR})
-        endif (UNIX AND (NOT APPLE))
+        if (WITH_DOC)
+           install(DIRECTORY ${SPHINX_PDF_DIR} DESTINATION ${CMAKE_INSTALL_DOCDIR})
+        endif (WITH_DOC)
     endif(PDFLATEX_FOUND)
     if (EXISTS ${QT_QCOLLECTIONGENERATOR_EXECUTABLE})
         add_custom_target( doc-qch-sphinx ${SPHINX_EXECUTABLE}
@@ -46,9 +48,9 @@ if(SPHINX_FOUND)
 		                   ${SPHINX_QCH_DIR}/*.qhcp
 				   DEPENDS doc-qch-sphinx )
         add_dependencies(doc doc-qch)
-        if (UNIX AND (NOT APPLE))
-	    install(DIRECTORY ${SPHINX_QCH_DIR} DESTINATION ${CMAKE_INSTALL_DOCDIR})
-        endif (UNIX AND (NOT APPLE))
+        if (WITH_DOC)
+            install(DIRECTORY ${SPHINX_QCH_DIR} DESTINATION ${CMAKE_INSTALL_DOCDIR})
+        endif (WITH_DOC)
     endif()
     add_custom_target( doc-html ${SPHINX_EXECUTABLE}
 	                        -q -c . -b html

--- a/doc/building.rst
+++ b/doc/building.rst
@@ -151,6 +151,10 @@ To build in installer (requires the mingw32-cross-nsis packages)::
 
   make package
 
+Known cmake parameters:
+
+* WITH_DOC=TRUE: create doc and manpages via running ``make``; also adds install statements to be able to install it via ``make install``.
+
 .. _`ownCloud repository from OBS`: http://software.opensuse.org/download/package?project=isv:ownCloud:devel&package=owncloud-client
 .. _CSync: http://www.csync.org
 .. _`Client Download Page`: http://owncloud.org/sync-clients/

